src/main/java/com/bandwidth/sqs/queue/buffer/task/Task.java

```diff
@@ -0,0 +1,18 @@
+package com.bandwidth.sqs.queue.buffer.task;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+public interface Task<K, T> {
```

Comment:
This interface and it's methods deserve some javadoc love, don't you agree? Especially the args are not fully self explaining.